L1Trigger/TrackFindingTracklet/plugins/ProducerKFin.cc

```diff
@@ -0,0 +1,261 @@
+      iEvent.getByToken<TTTracks>(edGetTokenTTTracks_, handleTTTracks);
+      const TTTracks& ttTracks = *handleTTTracks.product();
+      vector<TTTrackRefs> ttTrackRefsStreams(numStreamsTracks);
+      vector<int> nTTTracksStreams(numStreamsTracks, 0);
+      int channelId;
+      for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+        if (trackBuilderChannel_->channelId(ttTrack, channelId))
+          nTTTracksStreams[channelId]++;
+      channelId = 0;
+      for (int nTTTracksStream : nTTTracksStreams)
+        ttTrackRefsStreams[channelId++].reserve(nTTTracksStream);
+      int i(0);
+      for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+        if (trackBuilderChannel_->channelId(ttTrack, channelId))
+          ttTrackRefsStreams[channelId].emplace_back(TTTrackRef(handleTTTracks, i++));
+      for (channelId = 0; channelId < numStreamsTracks; channelId++) {
+        const TTTrackRefs& ttTrackRefs = ttTrackRefsStreams[channelId];
+        vector<TrackKFin> tracks;
+        tracks.reserve(ttTrackRefs.size());
+        vector<StubKFin> stubs;
+        const int nStubs = accumulate(ttTrackRefs.begin(), ttTrackRefs.end(), 0, [](int& sum, const TTTrackRef& ttTrackRef){ return sum += ttTrackRef->getStubRefs().size(); });
+        stubs.reserve(nStubs);
+        int trackId(0);
+        vector<int> numLayerStubs(setup_->numLayers(), 0);
+        for (const TTTrackRef& ttTrackRef : ttTrackRefs) {
+          // cut on more then 256 tracks
+          if (trackId >= dataFormats_->format(Variable::trackId, Process::kfin).range())
+            continue;
+          // get rz parameter
+          double cot = ttTrackRef->tanL();
+          double zT = ttTrackRef->z0() + setup_->chosenRofZ() * cot;
+          int binEta(-1);
+          for (; binEta < setup_->numSectorsEta(); binEta++) {
+            if (zT < sinh(setup_->boundarieEta(binEta + 1)) * setup_->chosenRofZ())
+              break;
+          }
+          // cut on outer eta sector boundaries
+          if (binEta == -1 || binEta == setup_->numSectorsEta())
+            continue;
+          cot -= setup_->sectorCot(binEta);
+          zT -= setup_->sectorCot(binEta) * setup_->chosenRofZ();
+          // cut on eta and |z0| < 15 cm
+          if (!dfZT.inRange(zT) || !dfCot.inRange(cot))
+            continue;
+          const int binZT = dfZT.toUnsigned(dfZT.integer(zT));
+          const int binCot = dfCot.toUnsigned(dfCot.integer(cot));
+          const vector<int>& layerEncoding = layerEncoding_->layerEncoding(binEta, binZT, binCot);
+          // get rphi parameter
+          double qOverPt = ttTrackRef->rInv() / 2.;
+          double phiT = deltaPhi(ttTrackRef->phi() - setup_->hybridChosenRofPhi() * qOverPt - ttTrackRef->phiSector() * setup_->baseRegion());
+          const int sectorPhi = phiT < 0. ? 0 : 1; // dirty hack
+          phiT -= (sectorPhi - .5) * setup_->baseSector();
+          // cut on nonant size and pt > 2 GeV
```

Comment:
These comments won't reflect any changes to cut parameters being changed so is potentially misleading, instead could have a comment pointing the user to where these parameters are being changed

Reply:
done